### ISSUE TITLE
Gas Report on reverted transactions

### ISSUE THREAD

**Issue by bh2smith** [cgewecke/hardhat-gas-reporter#53]
It appears the `chai` matching pattern is incompatible with gas reporting on reverted transactions. That is, in order for the gas report to come through we must use the the following pattern:

```js
const tx = (await contract.method()).wait();
````

Without the wait, the gas reporter doesn't recognize that the method has been called. This is fine for adjusting the unit test which expect transactions to succeed as in this example:

### OLD TEST
```js
    it("should add a solver", async () => {
      await expect(
        authenticator.connect(owner).addSolver(solver.address)
      ).to.not.be.reverted;
    });
```

### NEW TEST
```js
    it("should add a solver", async () => {
      const tx = await authenticator.connect(owner).addSolver(solver.address);
      await expect(tx.wait()).to.not.be.reverted;
    });
```

However, this adjustment does not work for tests that are expected to revert. For example, 

```js
    it.only("should not allow non-owner to add solver", async () => {
      await expect(
        authenticator.connect(nonOwner).addSolver(solver.address),
      ).to.be.revertedWith("caller is not the owner");
    });
 ```
 
For as soon as we set `const tx = await authenticator.connect(nonOwner).addSolver(solver.address);` the test will fail with "Error: VM Exception while processing transaction: revert Ownable: caller is not the owner"

Perhaps it is fine to only expect gas reports on successful transactions, just thought I would bring up this slight incompatibility here.

Furthermore, I suspect this to also be the case for view methods (since they do not result in actual transactions).

**Comment by cgewecke** [comment]
@bh2smith Thanks for reporting. Will investigate the the chai matching pattern problem - was not aware of that.

RE: measuring gas for reverted transactions - at the moment these are intentionally filtered out by checking the transaction status in the gas-reporter [code, here][1].

It would be simple to make this optional though... 

Out of curiosity, could you explain why you'd like to see gas usage data for transactions that fail?   

[1]: https://github.com/cgewecke/eth-gas-reporter/blob/bbf521f137407f106aeec44c730541dda6734ce3/lib/transactionWatcher.js#L43-L44

**Comment by bh2smith** [comment]
Hey! Thanks for getting back so quickly. Its not that we necessarily want to measure gas consumption on reverted transactions, but more about having a consistent pattern for writing unit tests. At the moment, I have worked around this by declaring and awaiting the transaction before the assertion followed by a `.wait()`. However, this would fail at the transaction declaration for those that are expected to revert. Of course, this simple workaround is fine for the time being especially considering the tests will essentially remain static forever after the contracts have been deployed. Just thought this might be the appropriate place to bring it up in the meantime. 

Also, a colleague of mine @fedgiac has proposed a few alternative workarounds [here](https://github.com/gnosis/gp-v2-contracts/pull/270#discussion_r544254047), but none seem to be entirely satisfactory...

**Comment by cgewecke** [comment]
Believe the bug you ran into is fixed in `1.0.4`. Opened an upgrade PR at `gp-v2-contracts` to double-check. 

The reporter was failing to track transactions sent with Waffle wallet accounts. 

Views and reverts are still excluded from the report, but everything else should be there. Please just lmk if you still see missing data.

**Comment by sebastiantf** [comment]
Is gas report for reverting transactions planned for the future? Would be nice to have that
